scan

```diff
@@ -330,16 +327,9 @@ def perform_scan(params: Tuple[Any, str, dict, dict, dict]):
             meta['duration'] = meta['end_time'] - meta['start_time']
 
         if data:
-            if hasattr(scanner, 'fast_cache_support'):
-                # Update the cache
-                if FAST_CACHE_KEY not in environment:
-                    environment[FAST_CACHE_KEY] = []
-                # I hate having to use isinstance here, but I don't
-                # see another way
-                if isinstance(data, list):
-                    environment[FAST_CACHE_KEY].extend(data)
-                else:
-                    environment[FAST_CACHE_KEY].append(data)
+            # Run the post-scan hook if it's present
+            if hasattr(scanner, 'post_scan'):
+                scanner.post_scan(domain, data, environment)
```

Comment:
Should the `post_scan` method also run even if `data` is `None`? I went through a bunch of scanners and couldn't find an example of a scanner returning `None` intentionally, but I would expect that even if there was an error in the interactions between the client and Lambda, that the ability to modify the fast cache (or take other post-scan actions) might be useful.

Reply:
This makes a lot of sense, especially since `post_scan()` is intended to be more general than just updating an in-memory cache.  It's a lifecycle method that may be used for any purpose whatsoever.

I'm committing a change now to allow fro it.